rts/Sim/Units/Unit.cpp

```diff
@@ -555,7 +562,7 @@ float3 CUnit::GetErrorVector(int argAllyTeam) const
 	const int atSightMask = losStatus[argAllyTeam];
 
 	const int isVisible = 2 * ((atSightMask & LOS_INLOS  ) != 0 ||                  teamHandler.Ally(argAllyTeam, allyteam)); // in LOS or allied, no error
-	const int seenGhost = 4 * ((atSightMask & LOS_PREVLOS) != 0 && gameSetup->ghostedBuildings && unitDef->IsBuildingUnit()); // seen ghosted building, no error
+	const int seenGhost = 4 * ((atSightMask & LOS_PREVLOS) != 0 && leavesGhost); // seen ghosted immobiles, no error
```

Comment:
Ghosts are no longer tied to being an immobile.
```suggestion
	const int seenGhost = 4 * ((atSightMask & LOS_PREVLOS) != 0 && leavesGhost); // seen ghost, no error
```

Reply:
Unsure about this, since this is just for ghosts of "generally immobile" units.

At least I don't see how this kind of ghosts could have a meaning in the game unless the unit has some kind of "generally immobile" assumption, even if it can sometimes be broken, like when a building is moved with some exception like being transported.

Mobile ones have a different kind of ghost.

Reply:
Can you point to the different kind of ghost for mobiles? Maybe handling for the two could be combined somehow.

Reply:
actually, looks like the mobile ones don't get drawn by engine itself, for bar it looks like it goes through unit_ghostradar_gl4.

Reply:
> actually, looks like the mobile ones don't get drawn by engine itself, for bar it looks like it goes through unit_ghostradar_gl4.

I suspect they could use the same path liveBuildingGhosts are using tho, they probably doing it for nothing